### PR TITLE
Test vercel-deploy workflow

### LINKED ISSUE #5557: Bot that publishes vercel branches on Teleport docs PRs
### What

We are using vercel in a creative way. Our repo `next` has multiple submodules to our docs. Once we merge the PR, the vercel build publishes the documents and updates the submodules. This works well, and allows us to have docs in sync with our branches (finally).

However, Vercel has a very cool feature, that is not available to us because of the way we use it. When creating a PR to repo, vercel triggers the build and posts a comment with preview. It makes it trivial to review the changes.

See for example here:

https://github.com/gravitational/next/pull/94#issuecomment-776874565

This is impossible to do with deploy hooks.

I wonder if we can:

* When someone submits a PR to one of our docs branches, create a PR to `next` repo that updates the submodule
* Use API to wait until vercel builds the branch and post the result to the original PR

Vercel has API and CLI:

https://vercel.com/download

https://github.com/vercel/vercel/blob/master/packages/now-client/src/deploy.ts

That would be fun project that will also help to improve UX

### PR DESCRIPTION
**Pull request by dumez-k**
This PR has been created to fix https://github.com/gravitational/teleport/issues/5557 

TODO:
- [x] Replace `VERCEL_TOKEN` with personal access token from @benarent 
- [ ]  Figure out why submodule updates not working
- [ ] Why isn't the Action Running

### CODE REVIEW

File: .github/workflows/vercel-preview.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+name: Deploy Vercel Preview
+env:
```
Comment (adaadb6) [inline .github/workflows/vercel-preview.yaml:2]: This section `env` can be removed.

File: .github/workflows/vercel-preview.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+on:
+  pull_request: 
+    paths: [ 'docs/**' ]
+    types: [ opened, reopened, edited, synchronize ]
+  workflow_dispatch:
+
+jobs:
+  deploy-vercel-preview:
+    name: Deploy vercel preview
+    runs-on: ubuntu-latest
```
Comment (adaadb6) [inline .github/workflows/vercel-preview.yaml:15]:
```suggestion
    runs-on: ubuntu-latest
    environment: vercel
```

File: .github/workflows/vercel-preview.yaml (hunk 1 above)
Comment (dumez-k) [reply .github/workflows/vercel-preview.yaml:2]: this section is defining environment variables necessary for the workflow

https://docs.github.com/en/actions/learn-github-actions/variables#defining-environment-variables-for-a-single-workflow

**Comment by dumez-k** [comment]
@wadells any idea why we can't see/manually trigger the workflow in the UI? Is there a permissions/security issue that new github actions workflows can't trigger on PR's?

**Comment by wadells** [comment]
> @wadells any idea why we can't see/manually trigger the workflow in the UI? Is there a permissions/security issue that new github actions workflows can't trigger on PR's?

You'll need a dummy copy of this workflow in the mainline before you can trigger it on a branch.

See the slack discussion here: https://gravitational.slack.com/archives/C08HF8E9F/p1692308817373059

**Comment by dumez-k** [comment]
> > @wadells any idea why we can't see/manually trigger the workflow in the UI? Is there a permissions/security issue that new github actions workflows can't trigger on PR's?
> 
> You'll need a dummy copy of this workflow in the mainline before you can trigger it on a branch.
> 
> See the slack discussion here: https://gravitational.slack.com/archives/C08HF8E9F/p1692308817373059

https://github.com/gravitational/teleport/pull/30652

File: .github/workflows/vercel-preview.yaml (hunk 1 above)
Comment (adaadb6) [reply .github/workflows/vercel-preview.yaml:2]: I don't think top level would work given the secrets are in the environment `vercel` (that get's referenced later on). You can also [move this to inside the job level](https://docs.github.com/en/actions/learn-github-actions/variables#defining-environment-variables-for-a-single-workflow) instead.

**Comment by ptgott** [comment]
Moving these commits over to the branch in this PR: https://github.com/gravitational/teleport/pull/30873

**Pull request closed without merging**